### PR TITLE
Enable ipv6 in orchagent

### PR DESCRIPTION
**Pull request by qiluo-msft**

### CODE REVIEW

File: orchagent/sainet.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
+// Header file definine base types and simple helper functions for network in SAI
```
Comment (stcheng) [inline orchagent/sainet.h]: rename to swssnet.h or just net.h  with namespace swss.

File: orchagent/sainet.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+#include <assert.h>
+#include <memory.h>
+#include <net/if.h>
+extern "C" {
+#include "sai.h"
+}
+#include "ipaddress.h"
+#include "ipprefix.h"
+
+inline static sai_ip_address_t& copy(sai_ip_address_t& dst, const swss::ip_addr_t& src)
```
Comment (stcheng) [inline orchagent/sainet.h]: are we using any ip_addr_t in our code? just move the content of this function into the copy(sai_ip_address_t, IpAddress). We shan't use ip_addr_t structure in our code except here in this file.

File: orchagent/sainet.h [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            memcpy(dst.addr.ip6, ia.ip_addr.ipv6_addr, 16);
+            memcpy(dst.mask.ip6, ma.ip_addr.ipv6_addr, 16);
+            break;
+        default:
+            assert(false); // unreachable code
+    }
+    return dst;
+}
+
+inline static sai_ip_prefix_t& copy(sai_ip_prefix_t& dst, const swss::ip_addr_t& src)
```
Comment (stcheng) [inline orchagent/sainet.h]: same as above

File: orchagent/sainet.h [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        case AF_INET:
+            dst.addr_family = SAI_IP_ADDR_FAMILY_IPV4;
+            dst.addr.ip4 = src.ip_addr.ipv4_addr;
+            break;
+        case AF_INET6:
+            dst.addr_family = SAI_IP_ADDR_FAMILY_IPV6;
+            memcpy(dst.addr.ip6, src.ip_addr.ipv6_addr, 16);
+            break;
+        default:
+            assert(false); // unreachable code
```
Comment (kcudnik) [inline orchagent/sainet.h]: asserts in release code "do nothing"
please replace with SWSS_LOG_ERROR("..") with explanation and then throw or exit(-1)

File: orchagent/sainet.h (hunk 1 above)
Comment (qiluo-msft) [inline orchagent/sainet.h]: Resolved

File: orchagent/sainet.h (hunk 2 above)
Comment (qiluo-msft) [inline orchagent/sainet.h]: resolved

File: orchagent/sainet.h (hunk 3 above)
Comment (qiluo-msft) [inline orchagent/sainet.h]: resolved

File: orchagent/sainet.h (hunk 4 above)
Comment (qiluo-msft) [inline orchagent/sainet.h]: resolved. It is actually a logic error (not a runtime error), and I will throw logic_error exception.

**Pull request merged**